packages/gridtools/package.py

```diff
@@ -37,8 +37,9 @@ class Gridtools(CMakePackage):
     depends_on('ncurses')
     depends_on('cmake@3.14.5:%gcc')
     depends_on('boost@1.67.0:')
-    depends_on('mpi',  type=('build', 'run'))
-    depends_on('cuda', when='+cuda', type=('build', 'run'))
+    depends_on('mpi',  type=('build', 'run'), when='~cuda')
+    depends_on('mpicuda',  type=('build', 'run'), when='+cuda')
+    depends_on('cuda', type=('build', 'run'), when='+cuda')
```

Comment:
I think they should also be link type. In gt2.0 it's irrelevant, but for gt1.x it should matter.

Reply:
If you think appending libcuda to LIBRARY_PATH is important then indeed yes